packages/ckeditor5-typing/src/inserttextobserver.ts

```diff
@@ -38,12 +39,20 @@ const TYPING_INPUT_TYPES = [
  * Text insertion observer introduces the {@link module:engine/view/document~Document#event:insertText} event.
  */
 export default class InsertTextObserver extends Observer {
+	/**
+	 * Instance of the focus observer. Insert text observer calls
+	 * {@link module:engine/view/observer/focusobserver~FocusObserver#flush} to mark the latest focus change as complete.
+	 */
+	public readonly focusObserver: FocusObserver;
+
 	/**
 	 * @inheritDoc
 	 */
 	constructor( view: View ) {
 		super( view );
 
+		this.focusObserver = view.getObserver( FocusObserver )!;
```

Comment:
It makes this class depend on the external order of initialization. Which is the reason of all the ugly `!` we were forced to add everywhere.
I'd rather add the paranoid check (throw an error) about not having `FocusObserver` in `View`.

Reply:
The `FocusObserver` is part of the `engine` package and always added by the `View`. The `InsertTextObserver` is in the plugin in the `typing` package so `FocusObserver` is always there.